app/kubemci/pkg/gcp/firewallrule/firewallrulesyncer.go

```diff
@@ -128,11 +138,33 @@ func (s *FirewallRuleSyncer) createFirewallRule(desiredFR *compute.Firewall) err
 	return nil
 }
 
+// Note: mutates the existingFR by clearing fields we don't care about matching.
 func firewallRuleMatches(desiredFR, existingFR *compute.Firewall) bool {
-	// TODO(nikhiljindal): Add proper logic to figure out if the 2 firewall rules match.
-	// Need to add the --force flag for user to consent overwriting before this method can be updated to return false.
 	// Also need to take special care of target tags. There can be multiple target tags, all of which can be "correct".
-	return true
+
+	// Clear output-only fields.
+	existingFR.CreationTimestamp = ""
+	existingFR.Id = 0
+	existingFR.Kind = ""
+	existingFR.SelfLink = ""
+	// Check for and clear default Direction.
+	if existingFR.Direction == "INGRESS" {
+		existingFR.Direction = ""
+	}
+	// Check for and clear the default Priority.
+	if existingFR.Priority == 1000 {
```

Comment:
hmm why is this required?
This will break if the default is changed. Should we care?

Reply:
The Firewall Rules that come back from gcp include the priority. We have 2 choices for dealing with this fact: 1) add P=1000 to our "desired" firewall rule; or 2) ignore P==1000 in our comparison. It seems pretty similar either way.
When the default changes:
with method 1), we'll be using the old default.
with method 2), we'll consider the rules not to match. And then until we fix the code, we'll always have to use --force, or a future "update" command.
So method (2) seems slightly better in that we won't use "old" defaults for anything. (Note that previous commits have been using method (1).)

Reply:
Another option is to ignore priority (dont compare it)?

Reply:
We discussed this in-person. I wrote up our thoughts in Issue#86. Basically, we'll move important attributes to the 'desiredBlah' function, and we'll decide what attributes can be ignored on a case-by-case basis, and put those into the BlahMatches function.
Pushing a new commit shortly.